### PR TITLE
Implemented blobstore interface in rust, wrote schema

### LINKED ISSUE #28: Create blobstore schema (All) and interface (Rust)
Create a Rust wrapper module around the blobstore contract schema

### PR DESCRIPTION
**Pull request by brooksmtownsend**
Going to lean on @autodidaddict a bit here to take a look over the schema, I took my best guess at appropriate `void` return types and parameters (e.g. using `Container` type for `container` field) so just looking for some validation this was done appropriately.

Fixes #28

### CODE REVIEW

**Review by autodidaddict — CHANGES_REQUESTED** [review]
The first thing I would say is attempt to go through a demo of a file system-based actor (we have some in `gantry` that we could use for examples) and see if this works.

Secondly, we should remove the host stuff that doesn't apply and remove the guest stuff that doesn't apply.

Guests can only register a `ReceiveChunk` handler, and all other functions in the interface are ones that are supplied by the host. For confirmation of this, you can look at the outdated capability provider code for the file system or S3, and you'll see in the descriptor code which way the functions go.

@pkedy we really need a way to describe the operation _direction_ in the interface so that when we generate, we don't have to continually re-delete the stuff that isn't supported.

File: rust/blobstore/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+// #[no_mangle]
+// pub fn wapc_init() {
+//     Handlers::register_create_container(create_container);
+//     Handlers::register_remove_container(remove_container);
+//     Handlers::register_remove_object(remove_object);
+//     Handlers::register_list_objects(list_objects);
+//     Handlers::register_upload_chunk(upload_chunk);
+//     Handlers::register_start_download(start_download);
+//     Handlers::register_start_upload(start_upload);
+//     Handlers::register_receive_chunk(receive_chunk);
```
Comment (autodidaddict) [inline rust/blobstore/src/lib.rs:24]: The only one of these registers that is applicable to a guest actor is `register_receive_chunk`

**Comment by autodidaddict** [comment]
@brooksmtownsend is this in the pipeline after graphdb?

**Comment by brooksmtownsend** [comment]
@autodidaddict Yep, this is on my todo after Graphdb. I was going to approach this in the same way, working to go through a demo of a filesystem actor to ensure we're still functional

File: rust/blobstore/src/lib.rs [hunk 2]
```diff
@@ -8,62 +8,61 @@
...
+//!
+//! #[no_mangle]
+//! pub fn wapc_init() {
+//!     http::Handlers::register_handle_request(download_poem);
+//!     blobstore::Handlers::register_receive_chunk(handle_chunk);
+//!     actor_core::Handlers::register_health_request(health);
+//! }
+//!
+//! /// Start the download of a blob (poem). Chunks will be streamed after download begins
+//! fn download_poem(req: http::Request) -> HandlerResult<http::Response> {
```
Comment (autodidaddict) [inline rust/blobstore/src/lib.rs:30]: :heart: :heart_eyes:

**Pull request merged**